bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/relay/RelayHandler.java

```diff
@@ -0,0 +1,278 @@
+     * <p>
+     * Two configurations are possible:
+     * 
+     * <ul>
+     * <li>Power Switch Mode (relay stays on indefinitely when switched on)</li>
+     * <li>Impulse Switch Mode (relay stays on for a configured amount of time and then switches off automatically)</li>
+     * </ul>
+     */
+    private void configureChannels() {
+        if (isInImpulseSwitchMode) {
+            configureImpulseSwitchModeChannels();
+        } else {
+            configurePowerSwitchModeChannels();
+        }
+    }
+
+    private void configureImpulseSwitchModeChannels() {
+        List<String> channelsToBePresent = List.of(CHANNEL_IMPULSE_SWITCH, CHANNEL_IMPULSE_LENGTH,
+                CHANNEL_INSTANT_OF_LAST_IMPULSE);
+        List<String> channelsToBeAbsent = List.of(CHANNEL_POWER_SWITCH);
+        configureChannels(channelsToBePresent, channelsToBeAbsent);
+    }
+
+    private void configurePowerSwitchModeChannels() {
+        List<String> channelsToBePresent = List.of(CHANNEL_POWER_SWITCH);
+        List<String> channelsToBeAbsent = List.of(CHANNEL_IMPULSE_SWITCH, CHANNEL_IMPULSE_LENGTH,
+                CHANNEL_INSTANT_OF_LAST_IMPULSE);
+        configureChannels(channelsToBePresent, channelsToBeAbsent);
+    }
+
+    /**
+     * Re-configures the channels of the associated thing, if applicable.
+     * 
+     * @param channelsToBePresent channels to be added, if not present already
+     * @param channelsToBeAbsent channels to be removed, if present
+     */
+    private void configureChannels(List<String> channelsToBePresent, List<String> channelsToBeAbsent) {
+        List<String> channelsToAdd = channelsToBePresent.stream().filter(c -> getThing().getChannel(c) == null)
+                .toList();
+        List<Channel> channelsToRemove = channelsToBeAbsent.stream().map(c -> getThing().getChannel(c))
+                .filter(Objects::nonNull).map(c -> Objects.requireNonNull(c)).toList();
+
+        if (channelsToAdd.isEmpty() && channelsToRemove.isEmpty()) {
+            return;
+        }
+
+        ThingBuilder thingBuilder = editThing();
+        if (!channelsToAdd.isEmpty()) {
+            addChannels(channelsToAdd, thingBuilder);
+        }
+        if (!channelsToRemove.isEmpty()) {
+            thingBuilder.withoutChannels(channelsToRemove);
+        }
+
+        updateThing(thingBuilder.build());
+    }
+
+    private void addChannels(List<String> channelsToAdd, ThingBuilder thingBuilder) {
+        for (String channelToAdd : channelsToAdd) {
+            Channel channel = createChannel(channelToAdd);
+            thingBuilder.withChannel(channel);
+        }
+    }
+
+    private Channel createChannel(String channelId) {
+        ChannelUID channelUID = new ChannelUID(getThing().getUID(), channelId);
+        ChannelTypeUID channelTypeUID = new ChannelTypeUID(BINDING_ID, channelId);
+        return ChannelBuilder.create(channelUID).withType(channelTypeUID).build();
```

Comment:
@jlaur is this the correct way to create channels dynamically?

Reply:
It looks good to me, so if it's also working, that's great. 😉 TBH, I'm not sure why there is a `create` overload taking `acceptedItemType` also, but it seems most other bindings use that.

Reply:
We need confirmation on this by @mike-bike whether this method actually works. Thanks for your input :+1: